src/resin-mixpanel-client.js

```diff
@@ -124,6 +124,34 @@ module.exports = function(token, options) {
 			} else {
 				throw new Error('(Resin Mixpanel Client) function identify is only available for the browser')
 			}
+		},
+		associateDeviceIds: function(deviceIds) {
+			if (!isBrowser) {
+				throw new Error('(Resin Mixpanel Client) function associateDeviceIds is only available for the browser');
+			}
+			var userName = mixpanel.get_property('$user_id');
+			var distinctId = mixpanel.get_property('distinct_id');
+			if (!userName) {
+				throw new Error('(Resin Mixpanel Client) mixpanel does not have correctly identified user info');
+			}
+			var originalDeviceId = mixpanel.get_property('$device_id');
+
+			// Send alias events to combine all input devices under a single user name.
+			for (var i = 0; i < deviceIds.length; i++) {
+				var deviceId = deviceIds[i];
+				mixpanel.register({
+					'distinct_id': deviceId,
+					'$device_id': deviceId
+				});
+				mixpanel.alias(userName);
```

Comment:
About the `alias` function, Mixpanel docs say:
> If the original ID is not passed in, we will use the current distinct_id -

So I guess this is what you are doing here: passing each `deviceId` with the `user_id`. Then to prevent a potential issue with the order of events here and play safe, shall we send the `deviceId` explicitly in this call?

Reply:
We set the distinct_id with `register` call 2 lines above. Hence, it does not really matter.